tests/framework/data/SqlDataProviderTest.php

```diff
@@ -32,7 +32,7 @@ public function testTotalCount()
             'sql' => 'select * from `customer`',
             'db' => $this->getConnection(),
         ]);
-        $this->assertSame(3, $dataProvider->getTotalCount());
+        $this->assertSame(3, (int) $dataProvider->getTotalCount());
```

Comment:
My opinion: your obsession of usage of `assetSame()` is go too far.
Not every comparison can be performed using it.